examples/ur5-ik.cpp

```diff
@@ -90,18 +90,19 @@ int main() {
       allocate_shared_eigen_aligned<QuadraticResidualCostTpl<Scalar>>(fn, w1);
   auto cost2 = allocate_shared_eigen_aligned<QuadraticDistanceCostTpl<Scalar>>(
       space, q0, w2);
-  auto cost = std::make_shared<CostSumTpl<Scalar>>(space->nx(), space->ndx());
+  auto cost = std::make_shared<CostSumTpl<Scalar>>(space.nx(), space.ndx());
```

Comment:
Use of shared pointer here. Do we plan to take care of it in this PR or the next?